src/libraries/System.Diagnostics.TraceSource/tests/System.Diagnostics.TraceSource.Config.Tests/TraceSourceWithConfigurationTests.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Diagnostics.TraceSourceConfigTests
+{
+    public class ConfigurationTests
+    {
+        private static volatile string? _configFile = null;
+
+        private static string ConfigFile
+        {
+            get
+            {
+                if (_configFile == null)
+                {
+                    Configuration.Configuration config = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+                    _configFile = Path.GetFileName(config.FilePath);
+                }
+
+                return _configFile;
+            }
+        }
+
+        private static void CreateAndLoadConfigFile(string filename)
+        {
+            Configuration.Configuration config = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+            string dir = Path.GetDirectoryName(config.FilePath);
+            string from = Path.Combine(dir, filename);
+            File.Copy(from, ConfigFile, overwrite: true);
+            TraceConfiguration.Register();
+            Trace.Refresh();
```

Comment:
The typical way a user would run this is add text to their config file and then call TraceConfiguration.Register() once at the beginning of their app without ever calling Trace.Refresh() right? Is there a test that validates that the configuration loads properly without invoking Trace.Refresh()?

Reply:
I'll look at adding that. Currently there is not a test that doesn't use Refresh(). This is because initialization is global, so we'd need a separate test suite just for that single test or detect first access before Refresh() and run that sanity test.

Reply:
Changed the test init code so the first test will not use Refresh().